### ISSUE TITLE
cannot import name 'get_color_name' from 'spyder_kernels.utils.nsview' 

### ISSUE THREAD

**Issue by liar666** [spyder-ide/spyder#15089]
### Issue Report Checklist

* [X ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ X] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [X ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [X ] Tried basic troubleshooting (if a bug/error)
    * [X ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [X ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

I'm trying to run spyder inside a fresh venv with lots of Data Science packages (numpy, scipy, matplotlib, scikit*, toch*, tf*).

I see the splash screen and very quickly I get error:

```
$ spyder
Traceback (most recent call last):
  File "/home/user/Python/DataScience/bin/spyder", line 8, in <module>
    sys.exit(main())
  File "/home/user/Python/DataScience/lib/python3.8/site-packages/spyder/app/start.py", line 213, in main
    mainwindow.main(options, args)
  File "/home/user/Python/DataScience/lib/python3.8/site-packages/spyder/app/mainwindow.py", line 3646, in main
    mainwindow = create_window(app, splash, options, args)
  File "/home/user/Python/DataScience/lib/python3.8/site-packages/spyder/app/mainwindow.py", line 3504, in create_window
    main.setup()
  File "/home/user/Python/DataScience/lib/python3.8/site-packages/spyder/app/mainwindow.py", line 791, in setup
    from spyder.plugins.console.plugin import Console
  File "/home/user/Python/DataScience/lib/python3.8/site-packages/spyder/plugins/console/plugin.py", line 29, in <module>
    from spyder.utils.environ import EnvDialog
  File "/home/user/Python/DataScience/lib/python3.8/site-packages/spyder/utils/environ.py", line 19, in <module>
    from spyder.plugins.variableexplorer.widgets.collectionseditor import (
  File "/home/user/Python/DataScience/lib/python3.8/site-packages/spyder/plugins/variableexplorer/widgets/collectionseditor.py", line 39, in <module>
    from spyder_kernels.utils.nsview import (
ImportError: cannot import name 'get_color_name' from 'spyder_kernels.utils.nsview' (/home/user/Python/DataScience/lib/python3.8/site-packages/spyder_kernels/utils/nsview.py)
```

Same error with `spyder --safe-mode`

### What steps reproduce the problem?

1. python3 -m venv DS && . ./DS/bin/activate 
2. python3 -m pip install -r requirements.txt # attached
3. spyder

Note: I'm running trough an `ssh -X` session.

### What is the expected output? What do you see instead?

spyder should open without error

### Paste Traceback/Error Below (if applicable)

Not applicable as spyder does not open

## Versions

* Spyder version:  [would be great to implement spyder --version...] spyder==4.2.5 / spyder-kernels==2.0.1
* Python version: Python 3.8.5 (default, Jan 27 2021, 15:41:15)  [GCC 9.3.0] on linux
* Qt version: 
* PyQt version:  PyQt5==5.15.4  / PyQt5-Qt5==5.15.2 [How did I managed to get 2 versions???]
* Operating System name/version: Linux Mint 20

### Dependencies

Spyder does not open

**Comment by ccordoba12** [comment]
> spyder==4.2.5 / spyder-kernels==2.0.1

Spyder 4 only works with spyder-kernels versions less than 2.0, as it's clearly declared in our `setup.py`:

https://github.com/spyder-ide/spyder/blob/4.x/setup.py#L226

So instead of running this command:

    python3 -m pip install -r requirements.txt # attached

please run instead

    python3  -m pip install spyder==4.2.5

instead.

**Comment by ccordoba12** [comment]
> PyQt5==5.15.4 / PyQt5-Qt5==5.15.2 [How did I managed to get 2 versions???]

This is wrong too because Spyder doesn't work with the latest version of PyQt5 yet. Our requirement is PyQt5 < 5.13

**Comment by f-seven** [comment]
I had the same error message after upgrading to Spyder 5.0 (Spyder 5.0, spyder_kernels 2.0.1, PyQt5 5.12.3).

After downgrading to 4.2.5 (`pip3 install spyder==4.2.5`), everything launches fine (Spyder 4.2.5 and spyder_kernels 1.10.3).

Running Mac OS 10.14.6 with Python 3.9.1.

**Comment by f-seven** [comment]
Tried upgrading to Spyder 5.0 on a different Mac running macOS 11.2.3 (BigSur) and Python 3.9.2 and spyder_kernels 2.0.1

Everything runs fine, no "cannot import name..." error message.

P.S. I do however get the following warning window:

`You have missing dependencies !`
`#Mandatory : spyder_kernes >=2.0.1, <2.1.0 : 2.0.1 (NOK)`

**Comment by ccordoba12** [comment]
Hey @f-seven, your problem is a duplicate of issue #15091 and it'll be fixed in our next version (5.0.1), to be released shortly.

@liar666, I'm going to close this issue because you're using an incorrect procedure to install Spyder (using the latest versions of its dependencies in a requirements file). Please use the command I posted above to install it correctly.
